Controllers/HomeController.cs

```diff
@@ -3,21 +3,88 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
+using System.Text.RegularExpressions;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Newtonsoft.Json;
 
 namespace Zoltu.Bags.Client.Controllers
 {
 	public class HomeController : Controller
 	{
+		private static Regex productRegx = new Regex(@"product\/(\d+)", RegexOptions.IgnoreCase);
+		private static Regex tagRegx = new Regex(@"tags\/(.*?)\/*?$", RegexOptions.IgnoreCase);
+		private static Regex minPriceRegx = new Regex(@"minprice\/(\d+)", RegexOptions.IgnoreCase);
+		private static Regex maxPriceRegx = new Regex(@"maxprice\/(\d+)", RegexOptions.IgnoreCase);
+		private static Regex aboutRegx = new Regex(@"aboutus", RegexOptions.IgnoreCase);
+		private static HttpClient client = null;
+
+		public HomeController()
```

Comment:
Adding Static Constructor wont help.
1- Readonly fields of regex cannot be initialized later, even in constructor.
2- If we move whole declaration then, fields will lose its scope.
3- Static constructor itself wont be effective, as declaring and initializing will keep a copy of initialized regex field in memory. which should be enough

As per https://www.dotnetperls.com/static
Static constructor. A static constructor initializes static fields. It runs at an indeterminate time before those fields are used. Static constructors on a type impose some overhead.